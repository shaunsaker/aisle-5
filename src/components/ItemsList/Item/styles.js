import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: styleConstants.dimensions.spacing.horizontal,
  },
  checkBoxContainer: {
    paddingRight: styleConstants.dimensions.spacing.horizontal,
  },
  deleteButtonContainer: {
    position: 'absolute',
    top: 0,
    right: styleConstants.dimensions.spacing.horizontal,
    bottom: 0,
    justifyContent: 'center',
  },
  shoppingItemContainer: {
    flex: 1,
    marginRight: styleConstants.dimensions.spacing.horizontal,
  },
});

export default styles;
