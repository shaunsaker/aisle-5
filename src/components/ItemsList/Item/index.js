import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import Animator from 'react-native-simple-animators';

import CheckBox from '../../CheckBox';
import ShoppingItem from '../../ShoppingItem';
import Counter from '../../Counter';

import styles from './styles';

const propTypes = {
  name: PropTypes.string,
  isChecked: PropTypes.bool,
  quantity: PropTypes.number,
  handleSetIsChecked: PropTypes.func,
  handleSetQuantity: PropTypes.func,
};

const defaultProps = {};

const Item = ({ name, isChecked, quantity, handleSetIsChecked, handleSetQuantity }) => {
  return (
    <View style={styles.container}>
      <View style={styles.checkBoxContainer}>
        <CheckBox isChecked={isChecked} handlePress={handleSetIsChecked} />
      </View>

      <View style={styles.shoppingItemContainer}>
        <ShoppingItem name={name} quantity={quantity} />
      </View>

      <Animator
        type="translateX"
        initialValue={0}
        finalValue={100}
        shouldAnimateIn={isChecked}
        shouldAnimateOut={!isChecked}
      >
        <Animator
          type="scale"
          initialValue={1}
          finalValue={0}
          shouldAnimateIn={isChecked}
          shouldAnimateOut={!isChecked}
        >
          <Counter value={quantity} handleChange={handleSetQuantity} />
        </Animator>
      </Animator>
    </View>
  );
};

Item.propTypes = propTypes;
Item.defaultProps = defaultProps;

export default Item;
