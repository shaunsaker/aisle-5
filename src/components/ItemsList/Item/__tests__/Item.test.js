import React from 'react';
import renderer from 'react-test-renderer';

import Item from '..';
import ITEM from '../../../../mockData/item';

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('Item', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Item {...ITEM} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the note checked state', () => {
      const component = renderer.create(<Item {...ITEM} isChecked={false} />);

      expect(component).toMatchSnapshot();
    });
  });
});
