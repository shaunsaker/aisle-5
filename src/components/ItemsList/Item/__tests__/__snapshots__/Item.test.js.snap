// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders renders the note checked state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingLeft": 15,
    }
  }
>
  <View
    style={
      Object {
        "paddingRight": 15,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#FFFFFF",
          "borderRadius": 15,
          "borderWidth": 1,
          "elevation": 6,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 2,
          "width": 30,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 0.01,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#FEA116",
                "fontSize": 24,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Muli-Bold",
        "fontSize": 17,
      }
    }
  >
    6
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "flex": 1,
        "fontFamily": "Muli-SemiBold",
        "fontSize": 17,
        "marginRight": 15,
      }
    }
  >
     x Beer
  </Text>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": null,
              "borderRadius": 30,
              "borderWidth": 0,
              "elevation": 6,
              "height": 30,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "#212121",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 2,
              "width": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#EDE8CA",
                    "fontSize": 30,
                  },
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#08293A",
                  },
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "width": 7.5,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": null,
              "borderRadius": 30,
              "borderWidth": 0,
              "elevation": 6,
              "height": 30,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "#212121",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 2,
              "width": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#EDE8CA",
                    "fontSize": 30,
                  },
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#08293A",
                  },
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Item renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingLeft": 15,
    }
  }
>
  <View
    style={
      Object {
        "paddingRight": 15,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#FFFFFF",
          "borderRadius": 15,
          "borderWidth": 1,
          "elevation": 6,
          "height": 30,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 2,
          "width": 30,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 0.01,
              },
            ],
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#FEA116",
                "fontSize": 24,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "fontFamily": "Muli-Bold",
        "fontSize": 17,
      }
    }
  >
    6
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#212121",
        "flex": 1,
        "fontFamily": "Muli-SemiBold",
        "fontSize": 17,
        "marginRight": 15,
      }
    }
  >
     x Beer
  </Text>
  <View
    style={
      Object {
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "right": 15,
        "top": 0,
        "transform": Array [
          Object {
            "translateX": 100,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": null,
            "borderRadius": 30,
            "borderWidth": 0,
            "elevation": 6,
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#212121",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.33,
            "shadowRadius": 2,
            "width": 30,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#EDE8CA",
                  "fontSize": 30,
                },
                Object {
                  "fontSize": 20,
                },
                Object {
                  "color": "#08293A",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 15,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": null,
              "borderRadius": 30,
              "borderWidth": 0,
              "elevation": 6,
              "height": 30,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "#212121",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 2,
              "width": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#EDE8CA",
                    "fontSize": 30,
                  },
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#08293A",
                  },
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "width": 7.5,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": null,
              "borderRadius": 30,
              "borderWidth": 0,
              "elevation": 6,
              "height": 30,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "#212121",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 2,
              "width": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#EDE8CA",
                    "fontSize": 30,
                  },
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#08293A",
                  },
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
